modules/seqexec/server/src/main/scala/seqexec/server/gpi/GPIController.scala

```diff
@@ -7,29 +7,137 @@ import cats.{Eq, Show}
+    LegacyObservingMode.UNBLOCKED_H    -> "H_unblocked",
+    LegacyObservingMode.UNBLOCKED_K1   -> "K1_unblocked",
+    LegacyObservingMode.UNBLOCKED_K2   -> "K2_unblocked"
+  )
+}
 
 final case class GPIController(gpiClient: GPIClient[IO]) {
   import GPIController._
+  import GPILookupTables._
   private val Log = getLogger
 
+  // scalastyle:off
   def gpiConfig(config: GPIConfig): SeqAction[CommandResult] = {
     val giapiApply = DefaultConfiguration
       .configurationBuilder()
+      .withConfiguration("gpi:selectAdc.deploy",
+                         (config.adc === LegacyAdc.IN)
+                           .fold(1, 0)
+                           .show)
+      .withConfiguration("gpi:configAo.useAo",
+                         config.aoFlags.useAo
+                           .fold(1, 0)
+                           .show)
+      .withConfiguration("gpi:configAo.useCal",
+                         config.aoFlags.useCal
+                           .fold(1, 0)
+                           .show)
+      .withConfiguration("gpi:configCal.fpmPinholeBias",
+                         (config.aoFlags.alignFpm)
+                           .fold(1, 0)
+                           .show)
+      .withConfiguration("gpi:configAo.optimize",
+                         config.aoFlags.aoOptimize
+                           .fold(1, 0)
+                           .show)
+      .withConfiguration("gpi:configIfs.integrationTime",
+                         (config.expTime.toMillis / 1000.0).show)
+      .withConfiguration("gpi:configIfs.numCoadds", config.coAdds.show)
+      .withConfiguration("gpi:configAo.magnitudeI", config.aoFlags.magI.show)
+      .withConfiguration("gpi:configAo.magnitudeH", config.aoFlags.magH.show)
+      .withConfiguration(
+        "gpi:selectShutter.calEntranceShutter",
+        (config.shutters.calEntranceShutter === LegacyShutter.OPEN)
+          .fold(1, 0)
+          .show)
       .withConfiguration(
-        "gpi:selectAdc.deploy",
-        (config.adc === edu.gemini.spModel.gemini.gpi.Gpi.Adc.IN)
+        "gpi:selectShutter.calReferenceShutter",
+        (config.shutters.calReferenceShutter === LegacyShutter.OPEN)
           .fold(1, 0)
           .show)
+      .withConfiguration(
+        "gpi:selectShutter.calScienceShutter",
+        (config.shutters.calScienceShutter === LegacyShutter.OPEN)
+          .fold(1, 0)
+          .show)
+      .withConfiguration(
+        "gpi:selectShutter.entranceShutter",
+        (config.shutters.entranceShutter === LegacyShutter.OPEN)
+          .fold(1, 0)
+          .show)
+      .withConfiguration("gpi:selectShutter.calExitShutter", "-1")
+      .withConfiguration("gpi:observationMode.mode",
+                         obsModeLUT.getOrElse(config.mode, ""))
+      .withConfiguration("gpi:selectPupilCamera.deploy",
+                         (config.pc === LegacyPupilCamera.IN)
+                           .fold(1, 0)
+                           .show)
```

Comment:
These are the things sent to the instrument itself? That is, `1` or `0`?

Reply:
yes we send 1 or 0